### PR TITLE
[Themes] Account for empty assets when calculating theme asset size

### PR DESCRIPTION
**Pull request by jamesmengo**
## WHY are these changes introduced?
Fixes https://github.com/Shopify/develop-advanced-edits/issues/339
Fixes https://github.com/Shopify/cli/issues/4470

### WHAT is this pull request doing?
- Add a fallback to calculate size when files are empty

### How to test your changes?
- Add an empty file to your theme via admin
- Run `shopify theme pull -t <THEME>`

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [x] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/cli-kit/src/public/node/themes/factories.ts [hunk 1]
```diff
@@ -52,7 +52,9 @@ export function buildThemeAsset(asset?: RemoteAssetResponse): ThemeAsset | undef
 
   const {key, checksum, attachment, value} = asset
   // Note: for attachments, this is the size of the base64 string, not the real length of the file
-  const stats = {size: (value || attachment).length, mtime: Date.now()}
```
Comment (frandiox) [inline packages/cli-kit/src/public/node/themes/factories.ts:55]: Oh, I thought they'd be empty string at this point. You can also just do:

```ts
  const stats = {size: (value || attachment || '').length, mtime: Date.now()}
```

**Review by frandiox — APPROVED** [review]

File: packages/cli-kit/src/public/node/themes/factories.ts (hunk 1 above)
Comment (frandiox) [reply packages/cli-kit/src/public/node/themes/factories.ts:55]: In fact, the types suggest these are already strings... I guess we are passing undefined as `any` directly from what we get from the cloud, thus bypassing TS checks?

Comment (karreiro) [reply packages/cli-kit/src/public/node/themes/factories.ts:55]: Indeed, possibly making `value` and `attachment` nullable in the `RemoteAssetResponse` would represent a bit better their meaning.

**Review by karreiro — APPROVED** [review]
Thank you for this PR, @jamesmengo! Great stuff :)

File: packages/cli-kit/src/public/node/themes/factories.ts (hunk 1 above)
Comment (jamesmengo) [reply packages/cli-kit/src/public/node/themes/factories.ts:55]: `size` is an empty string when the file is empty, so I changed this expression to `value ?? attachment` which returns `''`

`value || attachment` was returning `undefined`

**Pull request merged**